### ISSUE TITLE
Version is unclear

### ISSUE THREAD

**Issue by JojoIce** [dkimitsa/robovm-robopods#13]
First off, very nice to see all these robopods updated in one place!!!

Also, I am new to this area, so if I misunderstood the whole thing, feel free to close or move this to discussion or something. This is a mix of feedback of what possibly could be improved, possibly might be a bug and a humble request for some help.

For a newbie, it is very difficult to understand how the versions play together, and I think it is quite an important thing.

Looking at the root it shows a list of AltPods and they are for the "in dev version" of robovm-robopods of version 1.20.0 as I understand it. However, all released version seems to be a snapshot (even when they are released). Is there a reason for this mix of naming that one needs to understand? Google Mobile Ads (that I am interested in) seems to be baked into Firebase. In the root, I can see that for v 1.20.0 (or 1.20.0-SNAPSHOT) it should be Firebase v 8.12.1.

Navigating into the Firebase section 
https://github.com/dkimitsa/robovm-robopods/tree/alt/firebase
one can see mappings of older robopods versions to firebase versions, but there are also mappings of more pods (subpods?) where there is a mix of versions (Firebase versions?). So, for example, ios-mobile ads is version 9.0.0, but core is 8.12.1 and crashlytics is 8.12.0. It seems to be conflicting a bit with the previous page, or am I missing something?

A hint on the firebase page of where to get the older and newest versions would probably save a lot of time for newbies. 

In the page: ios-google-mobile-ads
https://github.com/dkimitsa/robovm-robopods/tree/alt/firebase/ios-google-mobile-ads
The commit messages that currently can be seen, seem to indicate firebase 8.12.1 and not 9.0.0 as stated in the firebase page or are there two different things that need to be installed/copied.
Trying to download the files and copy them in the right direction was tricky. Not sure I got it right. For example, specifying GoogleUtilities.framework but then pointing to a location where it actually was GoogleUtilities.xcframework, just two letters difference I just got that folder and nothing worked. I think it is supposed to be a subfolder somewhere under the GoogleUtilities.xcframework that should be copied but that was lost on this newbie. Maybe point to the exact folder that should be copied or refer to a general guide if there is one that can explain how this should be done.

The sample code on the page seems to be incorrect, probably outdated. It would be super nice with a small working sample code for banner, interstitial and rewarded if that exists or easily can be made or linked to or something. 

I went ahead with 1.20.0-SNAPSHOT (also tested 1.19.0-SNAPSHOT), and downloaded firebase-ios-sdk that I eventually found on github. Test v 8.10.0 through 8.12.1 as well as downloading from firebase itself that possibly was 9.0.0. 1.20.0-SNAPSHOT and 8.12.1 was the closest I got to work.

I get an error when I tried get feedback from rewarded ad or interstitial. I might very well have done something completely wrong here, since I am stumbling in the dark. It could be that I have mixed the files in the lib or using the wrong calls, but the error message a bit surprising.

So the critical part of my code is below:

```
public class AdsControllerIos {
	protected static final String REWARDED_AD_UNIT_ID = "ca-app-pub-3940256099942544/1712485313"; //Test id
	protected static final String TAG = "Ads init";
	private GADRewardedAd rewardAd;

	public AdsControllerIos() {
		GADMobileAds.sharedInstance().start(status -> Gdx.app.debug(TAG, "GAD init " + status));
		loadRewarded();
	}

	private void loadRewarded() {
		GADRewardedAd.load(REWARDED_AD_UNIT_ID, new GADRequest(), (GADRewardedAd gadRewardedAd, NSError error) -> {
			Gdx.app.debug(TAG, "GAD load rewarded error " + error);
			rewardAd = gadRewardedAd;
		});
	}

	@Override
	public void showRewardedVideoAd(final CallbackObject<Boolean> objectCallback) {
		if (rewardAd != null) {
			rewardAd.setFullScreenContentDelegate(new MyGADFullScreenContentDelegate("Rewarded AD")); //It goes all bad here
			rewardAd.present(UIApplication.getSharedApplication().getKeyWindow()
					.getRootViewController(), () -> objectCallback.doit(true));
			rewardAd.dispose();
			rewardAd = null;
		}
		loadRewarded();
	}
}
```
The class is created from the IOSLauncher and showRewardedVideoAd is called from the game sometime after the ad has been loaded. The MyGADFullScreenContentDelegate just implements the interface and logs some stuff:
```
public class MyGADFullScreenContentDelegate implements GADFullScreenContentDelegate {
	private String tag;

	public MyGADFullScreenContentDelegate(String tag) {
		this.tag = tag;
	}

	@Override
	public void didFailToPresentFullScreenContent(GADFullScreenPresentingAd ad, NSError error) {
		Gdx.app.debug(tag, "GAD interstitual callback " + error);
	}

	@Override
	public void adWillDismissFullScreenContent(GADFullScreenPresentingAd ad) {
		Gdx.app.debug(tag, "GAD interstitual callback adWillDismissFullScreenContent " + ad.toString());
	}

	@Override
	public void adDidRecordImpression(GADFullScreenPresentingAd ad) {
		Gdx.app.debug(tag, "GAD interstitual callback adDidRecordImpression " + ad.toString());
	}

	@Override
	public void adDidRecordClick(GADFullScreenPresentingAd ad) {
		Gdx.app.debug(tag, "GAD interstitual callback adDidRecordClick " + ad.toString());
	}

	@Override
	public void adDidDismissFullScreenContent(GADFullScreenPresentingAd ad) {
		Gdx.app.debug(tag, "GAD interstitual callback adDidDismissFullScreenContent " + ad.toString());
	}

	@Override
	public void adWillPresentFullScreenContent(GADFullScreenPresentingAd ad) {
		Gdx.app.debug(tag, "GAD interstitual callback adWillPresentFullScreenContent " + ad.toString());
	}
}
```

At the line where it states "It goes all bad here" I get the following:

java.lang.ClassCastException: com.trilligames.jump.ios.MyGADFullScreenContentDelegate cannot be cast to org.robovm.objc.ObjCObject
	at org.robovm.objc.ObjCObject$Marshaler.protocolToNative(ObjCObject.java:388)
	at org.robovm.pods.google.mobileads.GADRewardedAd.$m$setFullScreenContentDelegate$(Native Method)
	at org.robovm.pods.google.mobileads.GADRewardedAd.setFullScreenContentDelegate(GADRewardedAd.java)
	at com.trilligames.jump.ios.AdsControllerIos.showRewardedVideoAd(AdsControllerIos.java:118)

If I instead extends GADFullScreenContentDelegateAdapter, then it will not crash, it will actually show the ad, but I don't see any callback to indicate that user should be rewarded or that the ad is done showing (so that I can start the in game music again).

[debug] Rewarded AD: GAD interstitual callback adWillPresentFullScreenContent <GADRewardedAd: 0x282b77150>
[debug] Ads init: GAD load rewarded error null
Could not signal service com.apple.WebKit.WebContent: 113: Could not find specified service

I also tested roughly the same with GADRewardedInterstitialAd with the same result. The WebKit makes me think that I have missed something regarding what View/Controller to use but the class cast exception seems to be something quite different and maybe related to the lib/pod mapping.

Sorry for the big rant. Hopefully it can be of some use and If I have just made a silly mistake, then I'd be happy if someone could enlighten me.

**Comment by dkimitsa** [comment]
hi, thank you for interesting in alt-pods. 
few words how it works: 
- its my personal collection of pods maintains mostly for me and people around who is interested; 
- this repo doesn't accept any PRs -- if you need anything, open a bug. If you like to contribute -- better to do it to official page https://github.com/MobiVM/robovm-robopods 
- this repo is always in development state (SNAPSHOT), there is no release happening -- mostly due there is no testing phase attached. So all alt-pods here should be considered as not-tested;
- but there are still rules applies: once binding complete and dev branch is merged into `alt` -- there will be no major version change of libraries but bug-fixes mostly (if any). Any new framework/version updates will go to `dev-x` branch


regarding `java.lang.ClassCastException` -- what version of RoboVM do you use to compile it? 
as in version 2.3.13 there was a workaround introduced https://github.com/MobiVM/robovm/pull/563 that automatically converts Objects to NSObjects if these implement any NSObject protocol.
You should check your version or extend `MyGADFullScreenContentDelegate` from `NSObject`

**Comment by dkimitsa** [comment]
Forget about Firebase version mess :) 
version of pods targets version of Firebase iOS SDK, e.g. `1.20.0-SNAPSHOT` is binding of `8.12.1` as it available at release page https://github.com/firebase/firebase-ios-sdk/releases. 
`Firebase.zip` same time contains set frameworks each of them has own version (some match version of SDK some are not). You can check the version table in `Firebase.zip/README.md`
